src/coreclr/pal/src/misc/perfjitdump.cpp

```diff
@@ -2,15 +2,15 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // ===========================================================================
 
-#if defined(__linux__)
-#define JITDUMP_SUPPORTED
-#endif
-
 #include "pal/palinternal.h"
 #include "pal/dbgmsg.h"
 
 #include <cstddef>
 
+#if defined(__linux__) || (defined(__APPLE__) && defined(TARGET_OS_OSX) && TARGET_OS_OSX)
```

Comment:
```suggestion
#if defined(__linux__) || defined(__APPLE__)
```
Why is this not sufficient? It would be consistent with https://github.com/dotnet/runtime/pull/99986/files#diff-40b5cb5c723fa7b40fb7b8723df0e8ea7a5a3b1c593b602c1ab5930fc7296064R157-R159

Reply:
So I thought a bit more about your previous comment (about just enabling this unconditionally). The main issue is that on non-macOS, you can't write to `/tmp`. (I suppose you can in the simulator, but not on a device.) I could write files to `NSTemporaryDirectory()` which is inside the app's sandbox, but I don't know of any way to pull anything out of here other than by the app itself. So files written there wouldn't be useful, and there's no profiler that can profile on a device that understands jitdump/perfmap. Yet, maybe? I can file this as an enhancement for Instruments.

Does it make sense to still advertise the feature, going to `NSTemporaryDirectory()` on non-OSX & non-Simulator?

Reply:
> Does it make sense to still advertise the feature, going to NSTemporaryDirectory() on non-OSX & non-Simulator?

There is no CoreCLR build for non-macOS (yet), so this is a hypothetical question currently.

My comment was primarily about consistent use of ifdefs.